### PR TITLE
Add CSV Import to capcodes 

### LINKED ISSUE #327: CSV import for Aliasses
Hey, I'd recommend implementing CSV import for aliasses, the way with import.js has lots of possible errors and need a lot of manual work.
Implementing a CSV import where the end user can select which field goes to which and if existing aliasses for the same address should be ignored or overridden would be best.

### PR DESCRIPTION
**Pull request by DanrwAU**
# Description

Adds the ability to import CSV files for capcodes. 

Requires a header row containing at least address & alias all other fields are optional. 

There is currently no filtering for invalid headers and these are ignored (YOU HAVE BEEN WARNED!)

@pagermon/collaborators this one will need an actual review, I'm sure my code is dodgy to achieve what I have. 

Fixes #327 
## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Importing CSV's with correct header row works as expected
Trying to import CSV with no header row fails. 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated changelog.md with changes made

### CODE REVIEW

File: server/routes/api.js [hunk 1]
```diff
@@ -1007,33 +1007,137 @@ router.post('/capcodeRefresh', isLoggedIn, function(req, res, next) {
...
+          var color = capcode.color || 'black';
+          var icon = capcode.icon || 'question';
+          var ignore = capcode.ignore || 0;
+          var pluginconf = JSON.stringify(capcode.pluginconf) || "{}";
+          await db('capcodes')
+            .returning('id')
+            .where('address', '=', address)
+            .first()
+            .then((rows) => {
+              if (rows) {
```
Comment (SloCompTech) [inline server/routes/api.js:1063]: Add comment: Update alias with existing address

File: server/routes/api.js [hunk 2]
```diff
@@ -1007,33 +1007,137 @@ router.post('/capcodeRefresh', isLoggedIn, function(req, res, next) {
...
+                    })
+                  })
+                  .catch((err) => {
+                    importresults.push({
+                      address: address,
+                      alias: alias,
+                      result: 'failed' + err
+                    })
+                  })
+              } else {
```
Comment (SloCompTech) [inline server/routes/api.js:1089]: Add comment: Create new alias (address not in db)

File: server/routes/api.js [hunk 3]
```diff
@@ -1007,33 +1007,137 @@ router.post('/capcodeRefresh', isLoggedIn, function(req, res, next) {
...
     .catch((err) => {
       logger.main.error(err);
       return next(err);
     })
 });
 
+router.post('/capcodeImport', isLoggedIn, function (req, res, next) {
+  for (var key in req.body) {
+    //remove newline chars from dataset
+    req.body[key] = req.body[key].replace(/\r/g, '');
```
Comment (SloCompTech) [inline server/routes/api.js:1040]: Could here be: `req.body[key] = req.body[key].replace(/[\r\n]/g, '');` ?

Comment (DanrwAU) [reply server/routes/api.js:1040]: > req.body[key] = req.body[key].replace(/[\r\n]/g, '');

Does this replace both \r and \n individually?

Comment (SloCompTech) [reply server/routes/api.js:1040]: yes, as far as I remember it does (`[]` is for specifying a group of chars that match in regex), but better to test it to be sure

**Pull request merged**